Toplean_Alexandru/week-02/Domain/Domain/CreateMenuOp/CreateMenuCmd.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Domain.Models;
+
+namespace Domain.Domain.CreateMenuOp
+{
+    public struct CreateMenuCmd
+    {
+        public Restaurant Restaurant { get; }
+        public string Name { get; }
+        public MenuType MenuType { get; }
+
+        public CreateMenuCmd(Restaurant restaurant, string name, MenuType menuType)
+        {
+            Restaurant = restaurant;
+            Name = name;
+            MenuType = menuType;
+        }
+
+        // Returns a tuple. If everything is valid, the function will return true and the MenuErrorCode will be None.
+        // If it is not valid, then the function will return false with the known reason.
+        // Try-Catch block also added in case there is an unknown error and it will return false.
+        public (bool, MenuErrorCode) IsValid()
+        {
+            try
+            {
+                return HasIllegalCharacters(Name) ? (false, MenuErrorCode.IllegalCharacter) :
```

Comment:
Please only use conditional ternary operator (?:) when you want to evaluate a single condition, don't use imbrication constructions of ternary operator, it is error prone and can generate problems, in this case the old if-else statement is clear and easier to read